js/interface-lists.js

```diff
@@ -248,7 +251,24 @@ function attahObservers() {
 
         initFilePickerProvider(field);
       }
+    })
+    .on('change', '[name="image_type_select"]', function() {
+      var element = $(this);
+      var dataType = element.val();
+      var id = element.data('current-id');
+
+      switch (dataType) {
+        case 'all-folders':
+          selectedFoldersId.push(id);
+          break;
+          case 'url':
+            selectedFoldersId = _.filter(selectedFoldersId, function(item) {
```

Comment:
It works fine because Field 1 and Field 2 have different IDs which we consider. I better rename selectedFoldersId to selectedFieldId for clarification.